harness/determined/pytorch/_callback.py

```diff
@@ -75,3 +85,29 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         Load the state of this using the deserialized ``state_dict``.
         """
         pass
+
+
+class ClipGradsL2Norm(PyTorchCallback):
+    """
+    Callback that performs gradient clipping using
+    `L2 Norm <https://pytorch.org/docs/stable/nn.html#clip-grad-norm>`_.
+    """
+
+    def __init__(self, clip_value: float) -> None:
+        self._clip_value = clip_value
+
+    def on_before_optimizer_step(self, parameters: Generator) -> None:
```

Comment:
Let's make this an Iterator, not a Generator, since a Generator is more specific, and also not actually accurate here.

Also in other places.

Reply:
Their docs claim to return an iterator, but inspecting their source code they actually return  a Generator from `model.parameters()`. But happy to update this.